src/lib/y2storage/encryption.rb

```diff
@@ -322,16 +322,63 @@ def save_crypttab_name(devicegraph, entry)
 
         return unless device
 
-        if device.encrypted?
-          device.encryption.crypttab_name = entry.name
-        elsif entry.swap?
-          # Right now, a plain encryption device is created for swap only. Moreover, random password
-          # (urandom) is always considered, even when the crypttab file indicates another key file.
+        if create_swap_encryption_for_entry?(device, entry)
           device.remove_descendants
-          device.encrypt(dm_name: entry.name, method: :random_swap)
+          device.encrypt(dm_name: entry.name, method: encryption_method_for_entry(entry))
+        elsif device.encrypted?
+          device.encryption.crypttab_name = entry.name
         end
       end
 
+      # Whether the encrytion layer should be created for a swap device according to the given crypttab
+      # entry
+      #
+      # @param device [BlkDevice]
+      # @param entry [SimpleEtcCrypttabEntry]
+      #
+      # @return [Boolean]
+      def create_swap_encryption_for_entry?(device, entry)
+        encryption_method = encryption_method_for_entry(entry)
+
+        return false unless encryption_method_for_swap?(encryption_method)
+
+        encrypt_with_encryption_method?(device, encryption_method)
+      end
+
+      # Whether the given encryption method corresponds to a swap related encryption method
+      #
+      # @param encryption_method [EncryptionMethod]
```

Comment:
The type is `EncryptionMethod, nil`. I would say that's why you decided to have this method instead of just checking `#only_for_swap?`

Reply:
Yes, `nil` is a valid param value. And well, I prefer to have it as a separate method because I consider that it is different to `#only_for_swap?` from a semantic point of view. Maybe, in the future, we can "re-create" a swap encryption layer from a method that applies to swap but not only to swap.

Reply:
After "face to face" discussion, we have decided to move to `Crypttab` class all the logic related to `#save_crypttab_names`.